src/nix/installables.cc

```diff
@@ -409,6 +409,18 @@ InstallableFlake::getCursor(EvalState & state, bool useEvalCache)
     return res;
 }
 
+FlakeRef InstallableFlake::nixpkgsFlakeRef() const
+{
+    auto lockedFlake = lockFlake(*(cmd.getEvalState()), flakeRef, cmd.lockFlags);
```

Comment:
Calling `lockFlake` again is potentially expensive (and can print some warnings/messages again). So it would be better if we store `lockedFlake` somewhere in `InstallableFlake` so we only lock it once.

Reply:
Store it in _lockedFlake now